aten/src/ATen/core/ivalue.h

```diff
@@ -173,6 +174,17 @@ struct CAFFE2_API IValue final {
 
   // Tuple
   IValue(c10::intrusive_ptr<ivalue::Tuple> v);
+
+  template <typename Tuple, std::size_t... I>
+  IValue(const Tuple& t, c10::guts::index_sequence<I...>);
```

Comment:
Can we please make this constructor private?

Reply:
Also, we seem to have a polyfill for `std::apply` in `C++17.h`, can we just use that directly here?